### PR TITLE
Added new option in Settings page (closes #66)

### LINKED ISSUE #66: Add option to donate images
"as per the RFC I opened (https://discord.com/channels/781145214752129095/1060881403459407902), we have now switched to opt-in sharing images. You can now send "shared: true" in your requests, to make the horde share them with LAION. 

Remember that anonymous accounts automatically shares them! 

Also note that we will never share img2img or inpainting results!

@Stable Horde: Integrator: Please look into adding the switch for the new key and explain what it does. Furthermore, if you are reporting kudos costs for each generation, please adjust each image by +3 when share==False, and +1 when share==true. "

### PR DESCRIPTION
**Pull request by Khosbayar**
## Overview

Added new switch in Settings page that indicates if donate image option is enabled or not.

## Solution

Closes #66.

## Screenshot

Screenshot 2023-02-03 at 7 31 23 PM
Screenshot 2023-02-03 at 7 31 33 PM

### CODE REVIEW

**Comment by Khosbayar** [comment]
@ndahlquist I was doing that reverting `pubspec.lock` file. Did you just revert that file?

File: lib/blocs/tasks_bloc.dart [hunk 1]
```diff
@@ -58,6 +58,10 @@ class _TasksBloc {
           .replaceAll('{np}', ' ### $negativePrompt');
       print(formattedPrompt);
 
+      var isDonateImageEnabled = await sharedPrefsBloc.isDonateImageEnabled();
+      final shouldShare =
+          isDonateImageEnabled && img2ImgInputEncodedString == null;
```
Comment (Khosbayar) [inline lib/blocs/tasks_bloc.dart:63]: @ndahlquist Is there anything else I'm missing?

Comment (ndahlquist) [reply lib/blocs/tasks_bloc.dart:63]: Nope looks good!

**Comment by ndahlquist** [comment]
> I was doing that reverting pubspec.lock file. Did you just revert that file?

That commit was automatically created by a bot / github action. I'm guessing your Flutter instance might have accidentally locked outdated version of things; are you running the latest version of Flutter (3.7.0 or later)? It shouldn't be a big deal if not, just might explain the bot.

**Pull request merged**